core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -113,9 +113,24 @@ struct Rank {
 };
 
 namespace Impl {
+// Checked narrowing conversion that calls abort is the cast changes the value
+template <class To, class From>
+constexpr To checked_narrow_cast(From arg) {
+  constexpr const bool is_different_signedness =
+      (std::is_signed<To>::value != std::is_signed<From>::value);
+  auto const ret = static_cast<To>(arg);
+  if (static_cast<From>(ret) != arg ||
+      (is_different_signedness && (arg < From{}) != (ret < To{}))) {
```

Comment:
It should be enough to check that both of them are positive, shouldn't it? If one of the types is unsigned, values from that type cannot be negative anyway.